src/charm.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Charm the service.
+
+Refer to the following post for a quick-start guide that will help you
+develop a new k8s charm using the Operator Framework:
+
+https://discourse.charmhub.io/t/4208
+"""
+
+import logging
+import shutil
+
+import ops
+from charms.operator_libs_linux.v1 import snap
+from ops.framework import EventBase
+from ops.model import ActiveStatus, BlockedStatus
+from utils import safe_write_to_file
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+
+class KafkaBrokerRackAwarenessCharm(ops.CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+    def _on_install(self, event: EventBase):
+        """Handle on install event."""
+        cache = snap.SnapCache()
+        charmed_kafka = cache["charmed-kafka"]
+
+        # The charm should be deployed
+        if not charmed_kafka.present:
+            self.unit.status = BlockedStatus("Charmed kafka snap missing.")
+            event.defer()
+            return
+
+        self.unit.status = ActiveStatus()
+
+    def _on_config_changed(self, event):
+        """Handle config changed event."""
+        if not isinstance(self.unit.status, ActiveStatus):
+            event.defer()
+            return
+
+        rack_properties_file = "/var/snap/charmed-kafka/current/etc/kafka/rack.properties"
+        broker_rack = self.config["broker-rack"]
```

Comment:
**issue** can we just make a small validation of this? it should really never be `""`

Reply:
Setting `broker.rack=` should be valid on the properties, but we can guard against that if that's a problem.
Other than avoiding empty value, I don't know how we should validate this, as I think whatever random string could be the rack value no?